eox_nelp/signals/tests/test_receivers.py

```diff
@@ -64,3 +72,173 @@ def test_call_dispatch(self, dispatch_mock):
+        self.certificate_data = CertificateData(
+            user=UserData(
+                pii=UserPersonalData(
+                    username=self.username,
+                    email="harry@potter.com",
+                    name="Harry Potter",
+                ),
+                id=10,
+                is_active=True,
+            ),
+            course=CourseData(
+                course_key=self.course_key,
+            ),
+            mode="no-id-professional",
+            grade=5,
+            current_status="downloadable",
+            download_url="",
+            name="",
+        )
+        self.metadata = EventsMetadata(
+            event_type="org.openedx.learning.certificate.created.v1",
+            minorversion=0,
+        )
+
+    @override_settings(ENABLE_CERTIFICATE_PUBLISHER=False)
+    @patch("eox_nelp.signals.receivers.create_external_certificate")
+    def test_inactive_behavior(self, mock_task):
+        """Test that the asynchronous task wont' be called when the setting is not active.
+
+        Expected behavior:
+            - create_external_certificate is not called
+        """
+        certificate_publisher(self.certificate_data, self.metadata)
+
+        mock_task.delay.assert_not_called()
+
+    @patch("eox_nelp.signals.receivers.create_external_certificate")
+    def test_invalid_mode(self, mock_task):
+        """Test when the certificate data has an invalid mode.
+
+        Expected behavior:
+            - create_external_certificate is not called.
+            - Invalid error was logged.
+        """
+        invalid_mode = "audit"
+        log_info = (
+            f"The {invalid_mode} certificate associated with the user <{self.username}> and course <{self.course_key}> "
+            "doesn't have a valid mode and therefore its data won't be published."
+        )
+
+        certificate_data = CertificateData(
+            user=UserData(
+                pii=UserPersonalData(
+                    username=self.username,
+                    email="harry@potter.com",
+                    name="Harry Potter",
+                ),
+                id=10,
+                is_active=True,
+            ),
+            course=CourseData(
+                course_key=self.course_key,
+            ),
+            mode=invalid_mode,
+            grade=5,
+            current_status="downloadable",
+            download_url="",
+            name="",
+        )
+
+        with self.assertLogs(receivers.__name__, level="INFO") as logs:
+            certificate_publisher(certificate_data, self.metadata)
+
+        mock_task.delay.assert_not_called()
+        self.assertEqual(logs.output, [
+            f"INFO:{receivers.__name__}:{log_info}"
+        ])
+
+    @patch("eox_nelp.signals.receivers._generate_external_certificate_data")
+    @patch("eox_nelp.signals.receivers.create_external_certificate")
+    def test_create_call(self, mock_task, generate_data_mock):
```

Comment:
```suggestion
    def test_create_call(self, create_external_certificate_mock, generate_data_mock):
```
this is not more explicit?